src/stan/model/indexing/rvalue.hpp

```diff
@@ -249,8 +265,7 @@ inline auto rvalue(
     const char* name = "ANON", int depth = 0) {
   int m = idx.head_.n_;
   math::check_range("matrix[uni,multi] indexing, row", name, a.rows(), m);
-  Eigen::Matrix<scalar_type_t<EigMat>, 1, EigMat::ColsAtCompileTime> r
-      = a.row(m - 1);
+  auto&& r = a.row(m - 1);
```

Comment:
That will not work. Assigning a rvalue to rvalue reference will not extend lifetime of the rvalue so that `Eigen::Row` object will be destroyed after this line is executed, leaving a dangling reference. I was also wrong when I earlier replied to your question saying that `decltype(auto)` would work here. It would not for the same reason. Instead you have to do one of:
```suggestion
  const auto& r = a.row(m - 1);
```
```suggestion
  return rvalue(a.row(m - 1), idx.tail_);
```

Reply:
I like (3)!

Reply:
I was wrong again. While playing around with https://github.com/stan-dev/stan/pull/2885#discussion_r379287888, I found that [rvalue references also extend the lifetime of what they are pointing at](https://stackoverflow.com/questions/49371012/lifetime-of-rvalue-reference). So no change is necessary, even though I like suggestion 3.

Reply:
I'm fine with keeping it 3